blockly-core/core/ui/contract_editor/contract_editor.js

```diff
@@ -921,3 +943,23 @@ Blockly.ContractEditor.prototype.onPlaceExampleContent = function (currentY) {
 
   return newY;
 };
+
+Blockly.ContractEditor.prototype.onClose = function() {
+  if (!this.isOpen()) {
+    return;
+  }
+  var allPass = true;
+  this.exampleViews_.forEach(function (view) {
+    var failure = this.testExample(view.getBlock(), false);
+    view.setResult(failure);
+    view.refreshTestingUI(false);
+    allPass = allPass && !failure;
+  }.bind(this));
+  if (!allPass && this.appHijackedDialogClose_()) {
+    // app has taken responsibilty for closing dialog (likely by launching
+    // modal confirm dialog
+    return;
+  }
+
+  this.hideIfOpen.call(this);
```

Comment:
Heh. Probably not. I think my first approach had me sending params to hideIfOpen.